src/main/java/org/wildfly/security/auth/server/HttpAuthenticationFactory.java

```diff
@@ -49,7 +61,62 @@ HttpServerAuthenticationMechanism doCreate(final String name, final CallbackHand
     }
 
     Collection<String> getAllSupportedMechNames() {
-        return Arrays.asList(mechanismFactory.getMechanismNames(Collections.emptyMap()));
+        return asList(mechanismFactory.getMechanismNames(Collections.emptyMap()));
+    }
+
+    // TODO: at some point these should no longer be hard-coded
```

Comment:
And even more importantly we need to ensure we can support custom authentication mechanisms which will have their own set of supported credential and evidence types with algorithms.